utils-intervalrdd/src/main/scala/org/bdgenomics/utils/interval/array/IntervalArray.scala

```diff
@@ -108,6 +108,13 @@ object IntervalArray extends Serializable {
 trait IntervalArray[K <: Interval[K], T] extends Serializable {
   val array: Array[(K, T)]
   val maxIntervalWidth: Long
+  // maintains the last index returned from the most recent binarySearch result.
+  // !!!
+  // WARNING: NOT THREAD SAFE
+  // TO SAFELY USE THIS VARIABLE IN A SPARK CONTEXT, CREATE A SHALLOW COPY OF
+  // THE INTERVAL ARRAY.
+  // !!!
+  private var optLastIndex: Option[Int] = None
```

Comment:
@heuermh the issue is that we ship the `IntervalArray` as a broadcast object typically (we also use it inside an `IntervalRDD`, which guarantees a single reader per tree), which means it is shared across all of the threads running in a single executor. Each time you make a `get` call, it updates this `optLastIndex` value, so if you have multiple threads performing `get`s, you either need to synchronize the variable (expensive) or duplicate it across threads (by doing a shallow copy of the underlying array).

@devin-petersohn this documentation looks good, but the constraint around thread safety should also be mentioned in the public trait-level docs as it is important to users of the `IntervalArray` trait.